src/bin/pg_autoctl/cli_enable_disable.c

```diff
@@ -402,3 +432,593 @@ cli_disable_maintenance(int argc, char **argv)
+
+			/* update the Postgres SSL setup and maybe create the certificate */
+			if (!update_ssl_configuration(&postgres, mconfig.nodename))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_INTERNAL_ERROR);
+			}
+
+			/* update the monitor's configuration to use SSL */
+			if (!monitor_config_write_file(&mconfig))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_BAD_CONFIG);
+			}
+
+			/* display a nice summary to our users */
+			log_info("Successfully enabled new SSL configuration:");
+			log_info("  SSL is now %s",
+					 pgSetup->ssl.active ? "active" : "disabled");
+
+			if (pgSetup->ssl.createSelfSignedCert)
+			{
+				log_info("  Self-Signed certificates have been created and "
+						 "deployed in Postgres configuration settings "
+						 "ssl_key_file and ssl_cert_file");
+			}
+
+			log_info("  Health-check connections to Postgres nodes now "
+					 "use sslmode=%s",
+					 pgsetup_sslmode_to_string(pgSetup->ssl.sslMode));
+
+			break;
+		}
+
+		case PG_AUTOCTL_ROLE_KEEPER:
+		{
+			KeeperConfig kconfig = { 0 };
+			PostgresSetup *pgSetup = &(kconfig.pgSetup);
+			LocalPostgresServer postgres = { 0 };
+
+			bool updatedPrimaryConninfo = false;
+
+			kconfig.pgSetup = options.pgSetup;
+			kconfig.pathnames = options.pathnames;
+
+			if (!keeper_config_read_file(&kconfig,
+										 missingPgdataIsOk,
+										 pgIsNotRunningIsOk,
+										 monitorDisabledIsOk))
+			{
+				log_fatal("Failed to read configuration file \"%s\"",
+						  kconfig.pathnames.config);
+				exit(EXIT_CODE_BAD_CONFIG);
+			}
+
+			/* now override current on-file settings with CLI ssl options */
+			pgSetup->ssl = options.pgSetup.ssl;
+
+			local_postgres_init(&postgres, pgSetup);
+
+			/* log about the need to edit the monitor connection string */
+			(void) update_monitor_connection_string(&kconfig);
+
+			/* update the Postgres SSL setup and maybe create the certificate */
+			if (!update_ssl_configuration(&postgres, kconfig.nodename))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_INTERNAL_ERROR);
+			}
+
+			/* and write our brand new setup to file */
+			if (!keeper_config_write_file(&kconfig))
+			{
+				log_fatal("Failed to write the pg_autoctl configuration file, "
+						  "see above");
+				exit(EXIT_CODE_BAD_CONFIG);
+			}
+
+			/* update our primary_conninfo to the new sslmode, when secondary */
+			updatedPrimaryConninfo = update_primary_conninfo(&kconfig);
+
+			/* display a nice summary to our users */
+			log_info("Successfully enabled new SSL configuration:");
+			log_info("  SSL is now %s",
+					 pgSetup->ssl.active ? "active" : "disabled");
+
+			if (pgSetup->ssl.createSelfSignedCert)
+			{
+				log_info("  Self-Signed certificates have been created and "
+						 "deployed in Postgres configuration settings "
+						 "ssl_key_file and ssl_cert_file");
+			}
+
+			log_warn("  Postgres connection string to the monitor "
+					 "has not been changed, see above for details");
```

Comment:
This log seems unnecessary given all the info shown by `update_monitor_connection_string`.

```suggestion
```

Reply:
Actually never mind, I think this log is good now that I have seen it in the output.